profiles.js

```diff
@@ -1121,6 +1122,31 @@ module.exports = function (options) {
     }
   }
 
+  function cmd_ninjas_for_user(args, done) {
+    var seneca = this;
+    var userId = args.userId;
+    var ninjas = [];
+
+    if(args.user !== userId) return done(null, {ok: false, why: 'Invalid request'});
+
+    seneca.act({role: plugin, cmd: 'list', query: {userId: userId}}, function (err, profiles) {
+      if(err) return done(err);
+      if(_.isEmpty(profiles)) return done(null, ninjas);
+      var parentProfile = profiles[0];
+      if(_.isEmpty(parentProfile.children)) return done(null, ninjas);
+      async.each(parentProfile.children, function (ninjaUserId, cb) {
```

Comment:
this might be better written using async.map, e.g

```
 async.map(parentProfile.children, function (ninjaUserId, cb) {
        seneca.act({role: plugin, cmd: 'list', query: {userId: ninjaUserId}}, function (err, ninjaProfiles) {
          if(err) return cb(err);
          return cb(null, ninjaProfiles[0]);
        });
      }, function (err, ninjas) {
        if(err) return done(err);
        return done(null, ninjas);
      });
```
Also remove the `var ninjas = []` line, and also also the callback function can be replace with `done`